Add replication dashboards

francoisferrand opened the pull request:
- Add replication dashboards
- Add variable to select data/replay processor
- Allow injecting replication errors
- Allow configuring the MetricsConsumer groupId

![overview](https://user-images.githubusercontent.com/3909027/193259810-9cc409dc-b4c1-437d-8dfb-30015e2077d7.png)
![queue_populator](https://user-images.githubusercontent.com/3909027/193259821-46f3f905-bbf8-45d8-b1de-9eaba5bd62af.png)
![queue_processor](https://user-images.githubusercontent.com/3909027/193259836-3e26b866-1901-41a4-8493-f8add1130c73.png)
![replay](https://user-images.githubusercontent.com/3909027/193259845-33286bf7-0cd4-494d-988a-a0e7cccd108f.png)
![status](https://user-images.githubusercontent.com/3909027/193259852-a6b71382-16ba-4af0-82b1-15b7ffbab0a6.png)

Issue: BB-280

francoisferrand committed:
Add replication dashboards

Issue: BB-280

francoisferrand committed:
Add variable to select data/replay processor

Issue: BB-280

francoisferrand committed:
Allow injecting replication errors

This can be controlled by 2 variables:
* `BACKBEAT_INJECT_REPLICATION_ERROR_RATE` for the occurence rate;
* `BACKBEAT_INJECT_REPLICATION_ERRORS` to select the operations which
may be affected.

Issue: BB-280

francoisferrand committed:
Allow configuring the MetricsConsumer groupId

Issue: BB-280

bert-e commented:
# Hello francoisferrand,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand committed:
Fix the way read stage duration is collected

Issue: BB-280

francoisferrand committed:
Add queue-processor average stage duration

Issue: BB-280

francoisferrand committed:
Revert replication_stage_time_elapsed to milliseconds

It was unexpectedly changed to seconds, revert to millisecond for
consistency with other branches of backbeat: to have a single definition
of the metric.

Issue: BB-280

francoisferrand committed:
Fix success rate reduceCalc

Issue: BB-280

francoisferrand committed:
Cleanup heatmaps

Heapmap would sometimes display all slots when there is actually no data.

Issue: BB-280

francoisferrand committed:
Add replication alerts

Issue: BB-280

francoisferrand committed:
Add replication dashboard & alerts to image

Issue: BB-280

francoisferrand committed:
Add 'job' prefix to parameters

This improves consistency with the other dashboards & alerts.

Issue: BB-280

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-280 contains:


* `8.4.21`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.4.22`


Please check the `Fix Version/s` of BB-280, or the target
branch of this pull request.

francoisferrand committed:
Remove variable from threshold

While it works, the thresholds are not properly displayed in grafana:
thus it is better to use percentage "threshold" mode, and set the number
of replicas in the `max` field.

Issue: BB-280

francoisferrand committed:
Fix typo

Issue: BB-280

jbertran reviewed (approved):
LGTM

francoisferrand commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-280 contains:


* `8.4.22`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.4.23`


Please check the `Fix Version/s` of BB-280, or the target
branch of this pull request.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.4`



The following branches will **NOT** be impacted:

* `development/7.10`
* `development/7.4`
* `development/8.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.4`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/8.3`



Please check the status of the associated issue BB-280.

Goodbye francoisferrand.
